src/main/java/com/digitalasset/testing/ledger/DamlScriptRunner.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2019 Digital Asset (Switzerland) GmbH and/or its affiliates
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.digitalasset.testing.ledger;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+
+public class DamlScriptRunner {
+  private final Logger logger = LoggerFactory.getLogger(getClass().getCanonicalName());
+
+  private final ProcessBuilder processBuilder;
+
+  private Process script;
+
+  private DamlScriptRunner(ProcessBuilder processBuilder) {
+    this.processBuilder = processBuilder;
+  }
+
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  public void run() throws IOException, InterruptedException {
+    logger.debug("Executing: {}", String.join(" ", processBuilder.command()));
+    script = processBuilder.start();
+    if (script.waitFor() != 0) {
+      throw new IllegalStateException("Unexpected termination of DAML script.");
+    }
+    logger.info("DAML Script has run successfully.");
+  }
+
```

Comment:
```suggestion
    if (!scriptRunSuccessfully()) {
      throw new IllegalStateException("Unexpected termination of DAML script.");
    }
    logger.info("DAML Script has run successfully.");
  }

  private boolean scriptRunSuccessfully() throws InterruptedException {
    return script.waitFor(5, TimeUnit.MINUTES) && script.exitValue() == 0;
  }
```

Reply:
5 minutes is probably excessive. I remember there were some configured timeout values that could be used.

Reply:
I will pick something in between.